presto-main/src/main/java/com/facebook/presto/server/TaskResource.java

```diff
@@ -281,4 +284,10 @@ private static boolean shouldSummarize(UriInfo uriInfo)
     {
         return uriInfo.getQueryParameters().containsKey("summarize");
     }
+
+    private static Duration randomizeWaitTime(Duration waitTime)
+    {
+        long halfWaitMillis = waitTime.toMillis() / 2;
```

Comment:
Add a comment about why we randomize to [T/2, T] instead of making it symmetric around T (i.e., that since the client passes the wait time parameter it may be setting a timeout on its side)